analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -18,16 +18,55 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks.CSharp;
 
 public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 {
     private const string DiagnosticId = "S3900";
-    private const string MessageFormat = "Refactor this method to add validation of parameter '{0}' before using it.";
+    private const string MessageFormat = "{0}";
 
     internal static readonly DiagnosticDescriptor S3900 = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute() => true;
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        if (context.Operation.Instance.Kind == OperationKindEx.ParameterReference
+            && context.Operation.Instance.ToParameterReference() is { WrappedOperation: var reference, WrappedOperation.Syntax: var syntax, Parameter: var parameterSymbol }
```

Comment:
Capturing `syntax` makes the code harder to read and is unnecessary since it is only used once.

Reply:
I removed the capture for `WrappedOperation` and `WrappedOperation.Syntax`, but left it for `Parameter`, as it's referenced multiple times and I find it more readable to use an alias here.

Reply:
I would have kept `WrappedOperation`. If you don't want to capture it, I recommend introducing a local variable for `context.Operation.Instance`.